CONTRIBUTING.md

```diff
@@ -0,0 +1,417 @@
+- Build locally (`yarn build`) to catch MDX/TypeScript/Contentlayer errors.
+- Open a PR with a clear title, context, screenshots (if relevant), and a checklist (see below).
+
+## General Guidelines
+
+- Be complete and practical
+  - Cover end-to-end use cases. Link to related topics: ingestion, dashboard templates, alerts, query builder, and relevant features.
+  - Do not assume prior knowledge. Add brief context when needed and define terms on first use.
+  - Cross-link existing SigNoz docs instead of duplicating content. For example, when describing OTel Collector receivers or pipelines, reference the configuration guide: https://signoz.io/docs/collection-agents/opentelemetry-collector/configuration/
+- Be concise and direct
+  - Avoid filler and marketing fluff. Get to the point.
+  - Avoid generic intros like “In today’s digital landscape…” and adjectives like “powerful,” “robust,” “seamless.”
+- Prefer clarity over cleverness
+  - Use active voice and second person (“you”).
+  - Show before/after and expected outcomes when it helps.
+- AI/LLM Usage
+  - Using AI/LLMs for research is fine, but verify everything and rewrite in your voice.
+  - Do not paste unvetted AI/LLM text. Avoid vague generalities and ensure steps are reproducible.
+- Acronyms and short forms
+  - Define on first use, then use the short form consistently.
+  - Examples: “OpenTelemetry (OTel),” “OpenTelemetry Collector (OTel Collector),” “OpenTelemetry Protocol (OTLP).”
+- Placeholders and variables
+  - Use angle-bracket placeholders like `<service-name>`, `<region>`, `<SIGNOZ_INGESTION_KEY>`.
+  - Immediately below the snippet, explain what each placeholder means.
+
+
+## Content Structure
+
+Every doc should be skimmable and actionable.
+
+- Required frontmatter
+  - Always include and keep current:
+    ```yaml
+    ---
+    date: 2025-01-15           # YYYY-MM-DD
+    id: <unique-id-or-slug>
+    title: <Title in Sentence Case>
+    description: <1–2 line summary with key terms>
+    tags: [SigNoz Cloud, Self-Host]   # choose both if applicable
+    ---
+    ```
+  - Use `id` as a stable unique slug (no spaces); update links if it changes.
+  - Use `tags` consistently. Supported tags include `SigNoz Cloud` and `Self-Host`.
+- Standard sections (H2 level)
+  - `## Overview` – what the doc covers and when to use it.
+  - `## Prerequisites` – versions, accounts, keys, cluster access, etc. Include links.
+  - `## Steps` or specific setup sections – ordered, with subheadings for clarity.
+  - `## Validate`  – how to confirm it worked (UI path, endpoint, example output).
+  - `## Troubleshooting` – common issues and fixes (more details below).
+- Explaining commands and code
+  - Before each command, explain what it does and where to run it (local shell, container, Kubernetes, CI, etc.).
+  - After commands, note expected results and what happens next if relevant.
+  - For code/config blocks, annotate with language and filename to improve context:
```

Comment:
<img width="1014" height="203" alt="Image" src="https://github.com/user-attachments/assets/d44e3583-0b9c-48ce-a51c-9c11454318e5" />

in preview, it doesn't show that we have annotated with language - should we use a screenshot here to explain? Similarly for highlighting line 4 below. 

<img width="1029" height="235" alt="Image" src="https://github.com/user-attachments/assets/415e7e40-ec88-4fbf-8065-223663fe78ef" />